sunkit_pyvista/plotter.py

```diff
@@ -7,4 +15,126 @@ class SunpyPlotter:
     A plotter for 3D data.
 
     This class wraps `pyvsita.Plotter` to provide coordinate-aware plotting.
+    For now, all coordinates are converted to
+    a specific frame (`~sunpy.coordinates.HeliocentricInertial` by default),
+    and distance units are such that :math:`R_{sun} = 1`.
+
+    Parameters
+    ----------
+    coordinate_frame : astropy.coordinates.BaseFrame
+        Coordinate frame of the plot. The x, y, z axes of the pyvista plotter
+        will be the x, y, z axes in this coordinate system.
     """
+    def __init__(self, coordinate_frame=None):
+        if coordinate_frame is None:
+            coordinate_frame = HeliocentricInertial()
+        self._coordinate_frame = coordinate_frame
+        self._plotter = pv.Plotter(off_screen=True)
+
+    @property
+    def coordinate_frame(self):
+        """
+        Coordinate frame of the plot.
+        """
+        return self._coordinate_frame
+
+    @property
+    def plotter(self):
+        """
+        `pyvista.Plotter`.
+        """
+        return self._plotter
+
+    @functools.wraps(pv.Plotter.show)
+    def show(self, *args, **kwargs):
+        """
+        Show the plot.
+        """
+        self.plotter.show(*args, **kwargs)
+
+    def _coords_to_xyz(self, coords):
+        coords = coords.transform_to(self.coordinate_frame)
+        coords.representation_type = 'cartesian'
+        return np.column_stack((coords.x.to_value(R_sun),
+                                coords.y.to_value(R_sun),
+                                coords.z.to_value(R_sun)))
+
+    def _pyvista_mesh(self, m):
+        """
+        Create a mesh from a map.
+
+        Parameters
+        ----------
+        m : sunpy.map.Map
+
+        Returns
+        -------
+        pyvista.StructuredGrid
+        """
+        corner_coords = all_corner_coords_from_map(m)
+        nodes = self._coords_to_xyz(corner_coords.ravel())
+        grid = pv.StructuredGrid()
+        grid.points = nodes
+        grid.dimensions = [m.data.shape[0] + 1,
+                           m.data.shape[1] + 1,
+                           1]
+        data = m.data.T.reshape(-1)
+        grid['data'] = m.plot_settings['norm'](data)
+        return grid
+
+    def plot_map(self, m, **kwargs):
+        """
+        Plot a map.
+
+        Parameters
+        ----------
+        m : sunpy.map.Map
+            Map to be plotted.
+        **kwargs :
+            Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
+        """
+        cmap = kwargs.pop('cmap', m.cmap)
+        mesh = self._pyvista_mesh(m)
+        self.plotter.add_mesh(mesh, cmap=cmap, **kwargs)
+
+    def plot_line(self, coords, **kwargs):
+        """
+        Plot a line from a set of coordinates.
+
+        Parameters
+        ----------
+        coords : astropy.coordinates.SkyCoord
+        **kwargs :
+            Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
```

Comment:
```suggestion
        coords : `astropy.coordinates.SkyCoord`
            Coordinate Frame
        **kwargs :
            Keyword arguments are passed to `pyvista.Plotter.add_mesh`.
```
Maybe?

Reply:
Coordinates including the frame? Not just the frame right?

Reply:
I am not sure I follow.

Reply:
We pass a set of coordinates to `plot_line` so I thought coordinate frame might be misleading

Reply:
Hmm, I would add what you think is best then.